src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -874,6 +876,48 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return false;
 	}
 
+	/**
+	 * Refresh Azure access token if it's expired.
+	 * @param uri connection uri
+	 * @returns true if no need to refresh or successfully refreshed token
+	 */
+	public async refreshAzureAccountTokenIfNecessary(uri: string): Promise<boolean> {
+		let profile = this._connectionStatusManager.getConnectionProfile(uri);
+		if (!profile) {
+			this._logService.info(`Connection not found for uri ${uri}`);
+			return false;
+		}
+
+		//wait for the pending reconnction promise if any
+		let previousReconnectPromise = this._uriToReconnectPromiseMap[uri];
+		if (previousReconnectPromise) {
+			this._logService.info(`Found pending reconnect promise for uri ${uri}, waiting.`);
+			await previousReconnectPromise;
+			delete this._uriToReconnectPromiseMap[uri];
+		}
+
+		let expiry = profile.options.azureAccountTokenExpiresOn;
+		if (typeof expiry === 'number' && !Number.isNaN(expiry)) {
+			const currentTime = new Date().getTime() / 1000;
+			if (currentTime >= expiry) {
+				this._logService.info(`Access token expired for connection ${profile.id} with uri ${uri}`);
+				let newProfile = new ConnectionProfile(this._capabilitiesService, profile);
```

Comment:
Why are you creating a new connection profile here instead of just updating the existing one? Did you make sure that all the places the profile is cached is updated properly?

Reply:
I did this during debugging to get a clean profile but forgot to change it back. Updated!